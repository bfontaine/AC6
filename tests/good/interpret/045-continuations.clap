def return (v)   = fun (k) => k v
def bind (m) (f) = fun (k) => m (fun (a) => f a k)
def run (m)      = m (fun (x) => x)

def cpsfact (n) = 
  if (n = 0) then 
    return 1 
  else
    bind (cpsfact (n - 1)) (fun (r) => return (r * n))

val n = run (cpsfact 5)




