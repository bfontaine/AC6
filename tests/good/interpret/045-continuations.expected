:- return = <code>
:- bind = <code>
:- run = <code>
:- cpsfact = <code>
:- n = 120
